tests/test_loky_backend.py

```diff
@@ -677,3 +678,36 @@ def wait_for_handle(handle, timeout):
     if timeout is not None and timeout < 0.0:
         timeout = None
     return wait([handle], timeout)
+
+
+def _run_netesed_delayed(depth, delay, event):
+    if depth > 0:
+        p = multiprocessing.Process(target=_run_netesed_delayed,
+                                    args=(depth - 1, delay, event))
+        p.start()
+        p.join()
+    else:
+        event.set()
+
+    time.sleep(delay)
+
+
+def test_safe_terminate():
+    event = multiprocessing.Event()
+    p = multiprocessing.Process(target=_run_netesed_delayed,
+                                args=(4, 1000, event))
+    p.start()
+
+    # Wait for all the processes to be launched
+    if not event.wait(10):
+        safe_terminate(p)
+        raise RuntimeError("test_safe_terminate was not able to launch all "
+                           "nested processes.")
+
+    children = psutil.Process(pid=p.pid).children(recursive=True)
+    safe_terminate(p)
+
+    # The process can take some time finishing so we should wait up to 5s
+    gone, alive = psutil.wait_procs(children, timeout=5)
+    msg = "Should be no descendent left but found:\n{}"
+    assert len(alive) == 0, msg.format(children)
```

Comment:
Shouldn't that be `alive` instead of `children`?

```python
msg = "Should be no descendent left but found:\n{}"
assert len(alive) == 0, msg.format(alive)
```

Very nice test other than that.